File: Learn Programming/Learn Ruby/Command_Line_Argument.rb
# Check if arguments are provided
if ARGV.length == 0
  puts "No arguments provided. Usage: ruby script.rb <arg1> <arg2> ..."
  exit
end

# Accessing arguments
first_argument = ARGV[0]
second_argument = ARGV[1]

puts "First Argument: #{first_argument}"
puts "Second Argument: #{second_argument}"

puts "--------------------------------------"

# Iterate through all arguments
ARGV.each_with_index do |arg, index|
  puts "Argument #{index + 1}: #{arg}"
end

File: Learn Programming/Learn Perl/Command_Line_Argument.pl
#!/usr/bin/perl

# This Perl script demonstrates how to access command-line arguments.

# Usage: perl script.pl arg1 arg2 ...

# Get the number of command-line arguments
my $num_args = @ARGV;

# Print usage if no arguments provided
if ($num_args == 0) {
    print "Usage: perl $0 arg1 arg2 ...\n";
    exit;
}

# Print explanation of each argument
print "Explanation of command-line arguments:\n";
for (my $i = 0; $i < $num_args; $i++) {
    print "Argument $i: $ARGV[$i]\n";
}
